### ISSUE TITLE
ApolloServerPlugin#willSendResponse not propagate set headers - if there is Error

### ISSUE THREAD

**Issue by stengvac** [apollographql/apollo-server#5346]
Hello,
versions: `"apollo-server": "2.25.1"`

I have wrote ApolloServerPlugin which should set some headers to response in callback `willSendResponse` it works fine when there are not exceptions during plugin lifecycle. But if there is any `Error` then headers set like this

```
willSendResponse(context: GraphQLRequestContextWillSendResponse<ApolloCustomContext>) {
     context.response?.http?.headers?.set("myHeader", "someValue");
}
```

are not part of response. Method is invoked in both cases.

Expected: headers are propagated to response regardles if there are errors or not.

**Comment by glasser** [comment]
This will be fixed in Apollo Server 3.0 (by https://github.com/apollographql/apollo-server/pull/5287).

(We're tracking the release of 3.0 in https://github.com/apollographql/apollo-server/pull/5262 . Version `3.0.0-preview.2` is feature complete, but we haven't completed the migration guide and full doc refresh yet. That said, if you don't use a lot of the more obscure features of Apollo Server, the migration might be as simple as switching your dependency version, if you'd like to verify for me that this is fixed.  If you submit a full set of reproduction instructions that just required me to check out a repo or play with codesandbox.io to see the bug rather than having to be creative, I'd be happy to verify it against 3.0.0-preview.2 myself.)

**Comment by stengvac** [comment]
Thx for answer. I don't exactly know which Apollo features are counted as obscure so i will wait for official release.
